engine/tests/ElggCoreGetEntitiesTest.php

```diff
@@ -348,6 +348,38 @@ public function testElggAPIGettersTSPValidTypeValidSubtype() {
 	}
 
 	/**
+	 * Makes sure that ege* returns only entities of specified type,
+	 * when subtype in type_subtype_pairs set to an empty or false value,
+	 * or an array with a single false/null value.
+	 * See #8851
+	 */
+	public function testElggAPIGettersTSPReturnsOnlyEntitiesOfTypeForEmptyOrFalseSubtype() {
+		$types = $this->getRandomValidTypes(1);
+		$type = $types[0];
+
+		// Test no value subtype
+		$tests = array(
+			ELGG_ENTITIES_ANY_VALUE,
```

Comment:
Setting subtype to "any value" should return something, so this test needs a closer look. I'll try to do this soon.